packages/models-library/src/models_library/utils/string_substitution.py

```diff
@@ -0,0 +1,108 @@
+""" Utils for substitutions in string templates
+
+"""
+import re
+from collections import UserDict
+from string import Template
+from typing import Any
+
+OSPARC_IDENTIFIER_PREFIX = "OSPARC_ENVIRONMENT"
+
+
+def upgrade_identifier(identifier: str) -> str:
+    """
+    Converts legacy or incompatible identifiers into official replacement identifier
+    """
+    identifier = re.sub(r"[{}%]", "", identifier)
+    identifier = re.sub(r"[.-]", "_", identifier)
+    identifier = identifier.upper()
+    if not identifier.startswith(OSPARC_IDENTIFIER_PREFIX):
+        identifier = f"{OSPARC_IDENTIFIER_PREFIX}_{identifier}"
```

Comment:
You might also have some special cases like `SERVICE_VERSION` and `SIMCORE_REGISTRY` where this does not apply. You should also take these edge cases into consideration.

These cannot be changed, since they are already part of published services.

Reply:
I understand that. We will double check next PR when we bring the value sources